src/functions-and-arrays.js

```diff
@@ -1,41 +1,123 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else return num2;
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length === 0) return null;
+  if (words.length === 1) return words[0];
 
+  let result = words[0];
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length > result.length) {
+      result = words[i];
+    }
+  }
 
+  return result;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(numbers) {
+  // if (numbers === []) return 0;
 
+  let count = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    count += numbers[i];
+  }
+  return count;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
+function sum(numbers) {
+  let error = new Error("Unsupported data type sir or ma'am");
+
+  if (numbers.length === 0) return 0;
+  if (numbers.length === 1) return numbers[0];
```

Comment:
it's not necessary, in fact, this function could fail, if the array has only a single element like ["miami"], in this case you will return "miami" witch is not a valid output